mapviz_plugins/src/textured_marker_plugin.cpp

```diff
@@ -42,525 +37,483 @@
 #include <QGLWidget>
 
 // ROS libraries
-#include <ros/master.h>
-#include <sensor_msgs/image_encodings.h>
-
-#include <swri_math_util/constants.h>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/image_encodings.hpp>
 
 #include <mapviz/select_topic_dialog.h>
 
 // Declare plugin
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
+
+// C++ standard libraries
+#include <algorithm>
+#include <cmath>
+#include <cstdio>
+#include <map>
+#include <string>
+#include <vector>
+
 PLUGINLIB_EXPORT_CLASS(mapviz_plugins::TexturedMarkerPlugin, mapviz::MapvizPlugin)
 
 namespace mapviz_plugins
 {
-  TexturedMarkerPlugin::TexturedMarkerPlugin() :
-    config_widget_(new QWidget()),
-    is_marker_array_(false),
-    alphaVal_(1.0f) // Initialize the alpha value to default
+TexturedMarkerPlugin::TexturedMarkerPlugin()
+: MapvizPlugin()
+, ui_()
+, alphaVal_(1.0f)
+, config_widget_(new QWidget())
+, has_message_(false)
+{
+  ui_.setupUi(config_widget_);
+
+  // Set background white
+  QPalette p(config_widget_->palette());
+  p.setColor(QPalette::Background, Qt::white);
+  config_widget_->setPalette(p);
+
+  // Set status text red
+  QPalette p3(ui_.status->palette());
+  p3.setColor(QPalette::Text, Qt::red);
+  ui_.status->setPalette(p3);
+
+  QObject::connect(ui_.selecttopic, SIGNAL(clicked()), this, SLOT(SelectTopic()));
+  QObject::connect(ui_.topic, SIGNAL(editingFinished()), this, SLOT(TopicEdited()));
+  QObject::connect(ui_.clear, SIGNAL(clicked()), this, SLOT(ClearHistory()));
+  QObject::connect(ui_.alphaSlide, SIGNAL(valueChanged(int)), this, SLOT(SetAlphaLevel(int)));
+
+  // By using a signal/slot connection, we ensure that we only generate GL textures on the
+  // main thread in case a non-main thread handles the ROS callbacks.
+  qRegisterMetaType<marti_visualization_msgs::msg::TexturedMarker>("TexturedMarker");
+
+  QObject::connect(this,
+    SIGNAL(MarkerReceived(marti_visualization_msgs::msg::TexturedMarker)),
+    this,
+    SLOT(ProcessMarker(marti_visualization_msgs::msg::TexturedMarker)));
+}
+
+void TexturedMarkerPlugin::ClearHistory()
+{
+  RCLCPP_DEBUG(node_->get_logger(), "TexturedMarkerPlugin::ClearHistory()");
+  markers_.clear();
+}
+
+// TODO(P. J. Reed) could instead use the value() function on alphaSlide when needed,
```

Comment:
I feel contractually obligated to point out TODOs, even if they seem to be old.

Reply:
Someday I'll go back and revisit this, although it's also still in the ROS1 code, so probably better to do it with another PR that I can merge into both branches...